### DISCUSSION TITLE
Problem with Cotech weather station device decoder - No light data being received.

### DISCUSSION THREAD

**Discussion by VK1IAN** [merbanan/rtl_433#3099] [General]
I had a problem with compiling rtl_433 from source on a Pi model B running Debian 7 (Wheezy).
Christian helped me out (https://github.com/merbanan/rtl_433/discussions/3095) and although rtl_433 was working, I wasn't seeing Light and UV data from my Cotech 367959 weather station.
I was receiving this data previously and confirmed this by plugging in my dongle to my old laptop running version 21.12.
Figuring there was an issue with the cotech_36_7959.c driver, I replaced the one on my Pi (running the latest rtl_433) with the one from my 21.12 source tree and recompiled.
Bingo; I started to get Lux and UV readings.
I did a quick diff on the two versions of cotech_36_7959.c and it appears the newer version has some checking; apparently some models of the Cotech don't have the light sensor (mine does).
This is more of a FYI in case anyone else is not seeing light data and they are expecting to. Quick fix is to grab the Cotech_36_7959.c from an earlier version and recompile.
Programming is not my forte, but I'd like to have a crack at fixing it myself. I'll have a ferret through the code and see if I can figure out what is going on.
Stay tuned

**Comment by VK1IAN** [comment]
Well, I feel like a right idiot.
Before diving in head first, I decided to check the data being received given the comments I saw in the source file.
Lux was at 69627 and uv at 251 for the last 24hrs; appears as though my model Cotech does not have light sensors.
I'll close this discussion as user error.
Apologies for wasting your time.

**Reply by zuckschwerdt** [reply]
Every experiment and every knowledge gained will eventually help someone else, if shared!
